drivers/filesystems/udfs/create.cpp

```diff
@@ -774,26 +781,30 @@ UDFCommonCreate(
 
             if (TargetObjectName.Length != sizeof(FILE_ID)) {
                 AdPrint(("    Invalid file ID\n"));
+                ReturnedInformation = FILE_DOES_NOT_EXIST;
                 try_return(RC = STATUS_INVALID_PARAMETER);
             }
             Id = *((FILE_ID*)(TargetObjectName.Buffer));
             AdPrint(("  Opening by ID %8.8x%8.8x\n", (ULONG)(Id>>32), (ULONG)Id));
             if ((RequestedDisposition != FILE_OPEN) &&
                 (RequestedDisposition != FILE_OPEN_IF)) {
                 AdPrint(("    Illegal disposition for ID open\n"));
+                ReturnedInformation = 0;
                 try_return(RC = STATUS_ACCESS_DENIED);
             }
 
             RC = UDFGetOpenParamsByFileId(Vcb, Id, &TmpPath, &IgnoreCase);
             if(!NT_SUCCESS(RC)) {
                 AdPrint(("    ID open failed\n"));
+                ReturnedInformation = 0;
```

Comment:
Why not `FILE_DOES_NOT_EXIST` ?

Reply:
We can't really be sure of why the open by ID failed. Is it because of some other factors or the file not existing? Will make it ``FILE_DOES_NOT_EXIST`` if you recommend it.